File: src/commonMain/kotlin/io/kontainers/model/Container.kt
package io.kontainers.model

import kotlinx.serialization.Serializable

/**
 * Represents a Docker container with its properties.
 */
@Serializable
data class Container(
    val id: String,
    val name: String,
    val image: String,
    val state: ContainerState,
    val status: String,
    val ports: List<PortMapping>,
    val volumes: List<VolumeMount>,
    val networks: List<String>,
    val created: Long,
    val labels: Map<String, String> = emptyMap(),
    val env: List<String> = emptyList()
)

/**
 * Enum representing possible container states.
 */
@Serializable
enum class ContainerState {
    RUNNING, STOPPED, PAUSED, RESTARTING, REMOVING, DEAD, CREATED
}

/**
 * Represents a port mapping between container and host.
 */
@Serializable
data class PortMapping(
    val privatePort: Int,
    val publicPort: Int? = null,
    val type: String = "tcp",
    val ip: String = "0.0.0.0"
)

/**
 * Represents a volume mount between host and container.
 */
@Serializable
data class VolumeMount(
    val source: String,
    val destination: String,
    val mode: String = "rw"
)

File: v2/shared/src/models/container.ts
/**
 * Container-related models for Kontainers application
 */

/**
 * Represents a Docker container with its properties.
 */
export interface Container {
  id: string;
  name: string;
  image: string;
  state: ContainerState;
  status: string;
  ports: PortMapping[];
  volumes: VolumeMount[];
  networks: string[];
  created: number;
  labels?: Record<string, string>;
  env?: string[];
}

/**
 * Enum representing possible container states.
 */
export enum ContainerState {
  RUNNING = 'RUNNING',
  STOPPED = 'STOPPED',
  PAUSED = 'PAUSED',
  RESTARTING = 'RESTARTING',
  REMOVING = 'REMOVING',
  DEAD = 'DEAD',
  CREATED = 'CREATED'
}

/**
 * Represents a port mapping between container and host.
 */
export interface PortMapping {
  privatePort: number;
  publicPort?: number;
  type: string;
  ip: string;
}

/**
 * Represents a volume mount between host and container.
 */
export interface VolumeMount {
  source: string;
  destination: string;
  mode: string;
}